R/classicalmetaanalysis.R

```diff
@@ -105,1325 +80,3 @@ ClassicalMetaAnalysis <- function(jaspResults, dataset = NULL, options, ...) {
-.metaAnalysisSetError <- function(res, table) {
-  if(isTryError(res))
-    table$setError(.extractErrorMessage(res))
-}
-
-# Replaces "intrcpt" with "intercept" and concated "factorNamelevelName" with "factorName (levelName)"
-.metaAnalysisMakePrettyCoeffNames <- function(coeffNames, dataset, concatFactorNames = NULL) {
-  newNames <- coeffNames
-  
-  if (is.null(concatFactorNames))
-    concatFactorNames <- .metaAnalysisMapConcatFactorNames(dataset)
-  
-  for (i in seq_along(coeffNames)) {
-    coeffName <- coeffNames[i]
-    
-    if (coeffName == "intrcpt") {
-      newNames[i] <- "intercept"
-    } else if (!is.null(concatFactorNames)) {
-      coeffNameEnc <- .v(coeffName)
-      if (grepl(":", coeffNameEnc, fixed = TRUE)) { # it's an interaction term
-        terms <- unlist(strsplit(coeffNameEnc, ":", fixed = TRUE))
-        replaced <- .metaAnalysisMakePrettyCoeffNames(.unv(terms), dataset, concatFactorNames)
-        newNames[i] <- paste(replaced, collapse = "\u2009\u273b\u2009")
-      } else { # it's a regular term
-        match <- which(concatFactorNames == coeffName)
-        if (length(match) == 1)
-          newNames[i] <- names(concatFactorNames)[match]
-      }
-    }
-  }
-  
-  return(newNames)
-}
-
-# Creates a named character vector with values = "factorNameLevelName" and names = "factorName (levelName)"
-.metaAnalysisMapConcatFactorNames <- function(dataset) {
-  factorCols <- unlist(lapply(dataset, is.factor))
-  if (length(factorCols) == 0)
-    return(NULL)
-  
-  dataset <- dataset[factorCols]
-  
-  mapping <- levelsPerFactor <- lapply(dataset, levels)
-  names(levelsPerFactor) <- .unv(names(levelsPerFactor))
-  names(mapping) <- NULL
-  
-  for (i in seq_along(levelsPerFactor)) {
-    values <- paste0(names(levelsPerFactor)[i], levelsPerFactor[[i]])
-    values <- setNames(values, paste0(names(levelsPerFactor)[i], " (", levelsPerFactor[[i]], ")"))
-    mapping[[i]] <- values
-  }
-  
-  return(unlist(mapping))
-}
-
-# these functions originally lived in jaspBase/R/common.R, but are only used by this module
-as.modelTerms <- function(object, ...) UseMethod("as.modelTerms")
-as.modelTerms.list <- function(object) structure(object, class = "modelTerms")
-as.modelTerms.formula <- function(formula) structure(sapply(attr(terms(formula), "term.labels"), strsplit, ":"), class="modelTerms")
-formula.modelTerms <- function(modelTerms, env = parent.frame()) {
-  # Converts a modelTerms list into a one-side R formula
-  #
-  # Args:
-  #   modelTerms:  A list of interaction terms, each term being a list of variable names involved in the interaction
-  #   env:         An environement associated with the variables in the formula, see ?as.formula
-  #
-  # Value:
-  #   A formula. See ?formula
-  #
-  terms = sapply(modelTerms, function(x) paste0(unlist(x), collapse = ":"))
-  terms = terms[terms != ""]
-  formula.rhs = paste(terms, collapse = " + ")
-  if (formula.rhs != "") as.formula(paste(" ~ ", formula.rhs), env = env)
-}
```

Comment:
I guess these function should not be deleted but moved to classicalmetaanalysiscommon.R instead

Reply:
yes, that was it, thank you!